### PR TITLE
work with cycles

### PR DESCRIPTION
**Pull request by pavel-oleynik**

### CODE REVIEW

**Comment by RuslanFrankov** [comment]
hello

File: script.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
-alert ("Js it is ...");
-books = [];
-
+//alert ("Js it is ...");
+books =[];
```
Comment (RuslanFrankov) [inline script.js]:
```suggestion
books = [];
```
space

File: script.js [hunk 2]
```diff
@@ -22,22 +23,32 @@ function addBook() {
...
     let hasBook = false;
     let bookSave = document.getElementById('book-save');
     let bookBd = document.getElementById('book-bd');
-
+    
     for(let i = 0; i < books.length; i++) {
         let currentBook = books[i];
         if(book.id_number == currentBook.id_number) {
             hasBook = true;
+            books.push(book);
```
Comment (RuslanFrankov) [inline script.js]: we don't need to save (push) the same book again to database

File: script.js [hunk 3]
```diff
@@ -22,22 +23,32 @@ function addBook() {
...
             break;
         }
     }
+    //for(let i = 0; i < books.length; i++) {
+        if(book.id_number == books.id_number) {
```
Comment (RuslanFrankov) [inline script.js]: if we don't have this book in database , then we need to save it

File: index.html [hunk 4]
```diff
@@ -43,23 +44,24 @@ <h1>Books</h1>
...
         <br>
         <br>
+        <!--Button ADD, CLEAR FORM-->
         <button type="button" onclick="addBook();">Add</button>
         <button type="button" onclick="cleanFormInput();">Clear</button>
     </form>
 </div>
-
+      <!--Sorry-->
     <div id="book-save" style="visibility: hidden;"class="save-book save-book-monitor">
```
Comment (RuslanFrankov) [inline index.html]:
```suggestion
    <div id="book-save-error" style="visibility: hidden;"class="save-error save-book-monitor">
```
change style name accordinly

File: index.html [hunk 5]
```diff
@@ -43,23 +44,24 @@ <h1>Books</h1>
...
         <h2 class="text-danger-header">Sorry!</h2>
         <p class="text-danger"><strong>The book has already existed.</strong></p>
         <a id="book-save-click" href="#" class="button-book-save" value="" onclick="cleanButton();">Undestand!</a>
     </div>
- 
+    <!--Sucsess-->
     <div id="book-bd" style="visibility: hidden;"class="save-bd save-b-d">
```
Comment (RuslanFrankov) [inline index.html]:
```suggestion
    <div id="book-save-success" style="visibility: hidden;"class="save-success save-b-d">
```
change style name accordinly

File: script.js (hunk 3 above)
Comment (RuslanFrankov) [reply script.js:40]: create function to show Error Status
- show block with error message
- hide block with success message
- clear input

File: script.js (hunk 2 above)
Comment (RuslanFrankov) [reply script.js]: create function to show Error Status in other place and call it here

show block with error message
hide block with success message
clear input

Comment (RuslanFrankov) [reply script.js]: after you call showErrorStatus() function we return from addBook()

File: script.js [hunk 6]
```diff
@@ -46,29 +57,25 @@ function addBook() {
            document.getElementById('book-bd').style.visibility = "visible";
            //document.getElementById('form').style.opacity = "0";
            document.getElementById('book-save').style.visibility = "hidden";
```
Comment (RuslanFrankov) [inline script.js]: you can wrap this code in function showSuccessStatus() 
call it

File: script.js [hunk 7]
```diff
@@ -46,29 +57,25 @@ function addBook() {
...
+    addColumnToRow (rowTableArray, edition_year);
+    addColumnToRow (rowTableArray, author);
+    addColumnToRow (rowTableArray, cost);
+    addColumnToRow (rowTableArray, number_of_copies);
+    //document.querySelector('tableArray').appendChild(row)
+     function addColumnToRow(row, columnValue) {
+        var column = document.createElement("td");
+        column.innerHTML = columnValue;
+        row.appendChild(column);
     }
```
Comment (RuslanFrankov) [inline script.js]: it better to move this function out of addBook()

File: script.js [hunk 8]
```diff
@@ -46,29 +57,25 @@ function addBook() {
            document.getElementById('book-bd').style.visibility = "visible";
            //document.getElementById('form').style.opacity = "0";
            document.getElementById('book-save').style.visibility = "hidden";
+        }    
+        console.log(books);
+        console.log(`${title} = ` + 'the newbook is table');
+    let tableArray = document.getElementById('tableArray');
+    let rowTableArray = document.createElement("tr");
+    tableArray.appendChild(rowTableArray);
+    addColumnToRow (rowTableArray, id_number);
```
Comment (RuslanFrankov) [inline script.js]:
```suggestion
    addColumnToRow(rowTableArray, id_number);
```
no space is needed

File: script.js (hunk 6 above)
Comment (RuslanFrankov) [reply script.js]: and move this part to the end of function

File: script.js (hunk 1 above)
Comment (pavel-oleynik) [reply script.js]: done

File: script.js (hunk 2 above)
Comment (pavel-oleynik) [reply script.js]: done

**Review by RuslanFrankov — APPROVED** [review]
Good job!

**Pull request merged**